pkg/platform/platform.go

```diff
@@ -57,7 +58,7 @@ type Platform interface {
 	CreateFunction(ctx context.Context, createFunctionOptions *CreateFunctionOptions) (*CreateFunctionResult, error)
 
 	// EnrichFunctionConfig Enrich function config upon creating function
-	EnrichFunctionConfig(ctx context.Context, functionConfig *functionconfig.Config) error
+	EnrichFunctionConfig(ctx context.Context, functionConfig *functionconfig.Config, authSession auth.Session) error
```

Comment:
why not shoving the `authSession` to `ctx` itself? was there a specific reasoning not to?